### PR TITLE
1852 single dataset static predictions

### PR DESCRIPTION
**Pull request by TheSolY**
Pass pre-calculated predictions for single dataset checks to predictions for vision and to y_pred and y_proba for tabular instead of train_predictions etc.

### CODE REVIEW

File: deepchecks/vision/base_checks.py [hunk 1]
```diff
@@ -49,10 +51,12 @@ def run(
         random_state: int = 42,
         n_samples: Optional[int] = 10_000,
         with_display: bool = True,
+        predictions: Optional[Dict[int, Union[Sequence[torch.Tensor], torch.Tensor]]] = None,
```
Comment (JKL98ISR) [inline deepchecks/vision/base_checks.py:54]: add properties too

File: docs/source/user-guide/tabular/supported_models.rst [hunk 2]
```diff
@@ -121,6 +121,15 @@ ROC Curve and the AUC metric) will not run.
 Running Deepchecks with Pre-computed Predictions
 ------------------------------------------------
 
+It is possible to pass pre-computed predictions to the checks, this can be useful for example when the model is not
```
Comment (Nadav-Barak) [inline docs/source/user-guide/tabular/supported_models.rst:124]: Not sure this additional text is required, this info is IMO covered in the text above

File: docs/source/user-guide/tabular/supported_models.rst [hunk 3]
```diff
@@ -121,6 +121,15 @@ ROC Curve and the AUC metric) will not run.
...
+available, only the results previously saved from it.
+The pre-computed predictions should be passed to suite/check's ``run`` method in the appropriate format.
+The parameters to pass are ``y_pred`` and ``y_proba`` for single dataset checks or ``y_pred_train`` and
+``y_proba_train`` and ``y_pred_test`` and ``y_proba_test`` for checks that use both datasets.
+
+Code Example
```
Comment (Nadav-Barak) [inline docs/source/user-guide/tabular/supported_models.rst:130]: the additional title is redundant in my opinion

**Review by JKL98ISR — APPROVED** [review]

File: docs/source/user-guide/tabular/supported_models.rst (hunk 2 above)
Comment (noamzbr) [reply docs/source/user-guide/tabular/supported_models.rst:124]: Tend to agree. What is the added value of this paragraph?

**Review by noamzbr — COMMENTED** [review]
Looks good, agree with @Nadav-Barak though that the extra text in Running Deepchecks with Pre-computed Predictions seems unnecessary

**Review by noamzbr — APPROVED** [review]
Minor ammendment

**Pull request merged**